### ISSUE TITLE
Keener Communications (Now Nexa) - Terminate/Update SA Billing Services

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1525]
In GitLab by @kdjstudios on Jul 11, 2019, 11:07

**Submitted by:** Gaylan Garrett <Gaylan.Garrett@Nexa.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-07-10-33147/conversation - Closed

**Helpdesk Notice:**  http://www.servicedesk.answernet.com/profiles/ticket/2020-03-19-24586/conversation

**Server:** External

**Client/Site:** Keener

**Account:** NA

**Issue:** 

I hope all is well. Please accept this as official notice to terminate the SA billing account for Keener Communications. I have two requests and Kyle has already entered tickets / sent emails for both but I just wanted to mention them here.  We would like to have the client portal disabled asap so they do not make their payments on the SA billing platform since the invoices are now on the NEXA billing platform.

 

Secondly, if possible, we would like to have access to historical data at least until the end of 2019 at a discounted rate since we will no longer be using the billing platform to process invoices but occasionally may need to review the historical data for evaluating clients’ usage amounts. Looking forward to hearing from you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 11, 2019, 11:11

Keener has officially posted their 30 day notice to terminate services with us. David is working with their team on the termination process, but as stated above they would still like to have historical access. I would like to confirm that this could be accomplished by a Read Only security group, correct?

Also, once they have officially terminated (end of 2019?) we will need to terminate all users and access accordingly. in #1524 we have already restricted their client access accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 12, 2019, 12:23

Gaylan wrote:

> I have the ok to keep it minimally active until the end of the year.  So, 8/10 will be last day I will have full functuality of the system and I would guess the 8/1 invoice will have pro rate of 10 days of the 150.00 rate and 21 days of the 100 rate for about a 115.00 invoice for August.  After 8/10, I will be able to still run reports and have access to data, especially historical usage data but no capabilities to process billings or payments. Is this correct ?

@smasih @amishra @pchaudhary @tim.traylor 

I just want to confirm what security group we need to assign to their users?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 15, 2019, 01:50

@kdjstudios 


According to my best understanding, the security group you are asking for is: **Read Only**

![image](/uploads/c6ef9766189799331dda08906e59f778/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2019, 11:34

@pchaudhary 

Thank you for confirming which one. May we also have a list of all (Local/NON-LDAP) users who have access to the Keener sites, please. As of 8/11/19 we will want to switch all of those users over to the Read Only access.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 16, 2019, 23:56

@kdjstudios 

Thank you for the update.

We have exported of all local users for all keener sites into the CSV. Please find below attachment. 

[List of all users for keener sites](/uploads/1fffe7256f59715c551229ee898d6609/keener_user_lists3.csv) 

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 17, 2019, 12:43

Thank you. I will set a reminder in my calendar to edit those users on the 11th.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 22, 2019, 08:20

@tim.traylor @amishra @pchaudhary 

Since my alert was on the 11th, which is a Sunday, I did not see the alert. I just now went in an moved all users to read only status.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 23, 2019, 11:01

closed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2019, 11:50

reopened

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2019, 11:55

Reopened ticket, http://www.servicedesk.answernet.com/profiles/ticket/2019-12-17-93843

We will need to remove all Keener logins from the system the first week of 2020. I believe I should be able to do that without issue.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2019, 11:56

assigned to @kdjstudios

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2019, 13:17

After communicating with Gaylan, we are going to keep their access until the end of Feb. 2020.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 18, 2020, 15:40

I sent a draft email for review by Terri to be sent out to confirm their termination of services.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 20, 2020, 14:06

We sent out the confirmation letter that Keener will terminate at the end of March.

So far we have not heard back. So on April 1st we will need to have the Aavaz team disable access to SA Billing for all  Keener only user logins. I think we can do this by simply deleting their user logins correct?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 23, 2020, 04:28

@kdjstudios @jredfield 

Thank you for the update.

* Yes Kyle's you are right we will remove/destroy the **Local/NON- LDAP** user detail from the SAB database.

* If you want to destroy the **LDAP USER** then give me the list of all the users. So that we will remove the LDAP user from the SAB database as well.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 23, 2020, 10:28

@amishra 

I believe you have already provided that list in  http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1528#note_44063

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 23, 2020, 14:53

@kdjstudios @jredfield 

Yes but we have given only non ldap user.
You have to give list of all LDAP user.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 23, 2020, 16:22

@amishra 

Only non-ldap users should be removed as those are actual client users. LDAP users our on internal employees and should not be deleted.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 23, 2020, 23:32

@kdjstudios @jredfield 

Thank you for the update.

We have 26 users on the above-given listing. Once you will confirm we will remove these records from the SAB database.  

I request you please have a look at the [sheet](/uploads/bab5e40c94ad253a4b3aced7da74cf1c/keener_user_lists3__2_.csv).

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 24, 2020, 17:07

@amishra 

Please remove them on April 1st. Please make sure as requested to only remove users that have access to only the Keener sites and are non-ldap.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 24, 2020, 17:07

closed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 1, 2020, 08:37

reopened

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 1, 2020, 08:38

@amishra @tim.traylor 

Please remove the logins overnight. 

NOTE: I must have closed this by accident.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 1, 2020, 08:47

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 6, 2020, 04:19

@kdjstudios @tim.traylor 

Thank you for the update.
* We have removed all the non -LDAP users who have keener site access.
* Please find the below user lists.
[final_keener.csv](/uploads/87e3bae46669f37d7f9ca70f7e3de4ee/final_keener.csv)

* Also we noticed that two users have other site access along with keener site.
   * nikhilesh
   * tgentry2
* So, please let us know if we need to do for those two users.
EDIT:
* I noticed that keener has many clients. So do we need to remove the client as well.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2020, 09:11

@amishra 

Thank you for the update. 

@tim.traylor @Robleski 

I noticed that keener has many clients. So do we need to remove the client as well.
> This is a great point, and one that we missed reviewing originally. How do we want to handle Keener's client logins? I presume the same and we will want to remove them from the system. But please confirm/approve.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 09:13

Marking for discussion on Thursday

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 16, 2020, 09:20

This was discussed today. Please back up the collection and then delete the logins.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Apr 20, 2020, 08:34

@kdjstudios  @tim.traylor 

We run the script and find that there is no client associated with a keener so we don't need to delete any. So we can close the ticket.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 21, 2020, 08:35

@rranjan @tim.traylor @Robleski 

May someone please double check this. As Keener certainly did have client portal logins and to my knowledge there is no way for users to delete the logins from the UI. So I find it highly incorrect that there are no clients. 

I ran the portal payment report to see which accounts are reporting payments by client logins. I am sure there are more then just the accounts/customers listed on this report too... [portal_payment_04_21_2020.csv](/uploads/e38bfde58c000446a2d89de5d14ac541/portal_payment_04_21_2020.csv)

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Apr 22, 2020, 04:17

@kdjstudios  @tim.traylor 

We again looked into it and following is our finding:- 

1. The payment portal is showing record because they are coming from the credit card transaction table for a specific time period. 

2. We find a ticket on which there was the request to block client, please refer #1524

3. We run the script for all clients but we don't find any client associated with a keener site. Please find the CSV for detail.

[customer_with_client.csv](/uploads/5e2e8847be8e3050c098659e79888025/customer_with_client.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 22, 2020, 13:06

@rranjan @ranand @amishra @tim.traylor @Robleski 

May someone please explain then how we have payments from client logins,but no client logins?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 22, 2020, 23:41

@kdjstudios @tim.traylor 

Thank you for the update.

Let me explain here:

* We have removed the keener client portal login credentials 9 months ago on ticket #1524.
* All the Keener client payment before 9 months.
* While removing the Keener users we had forgotten that we have already removed the client portal for keener site.
* So we can close this ticket because there is no longer user & client for the keener site.

Thanks
